Sledge.Packages/Wad/WadPackage.cs

```diff
@@ -11,23 +12,26 @@ namespace Sledge.Packages.Wad
     // https://developer.valvesoftware.com/wiki/WAD
     public class WadPackage : IPackage
     {
-        private const string Signature = "WAD3";
+        private const string WAD2Signature = "WAD2";
+        private const string WAD3Signature = "WAD3";
 
         public FileInfo PackageFile { get; private set; }
         internal uint NumTextures { get; private set; }
         internal uint LumpOffset { get; private set; }
         internal List<WadEntry> Entries { get; private set; }
+        public Palette Palette { get; private set; }
 
-        public WadPackage(FileInfo packageFile)
+        public WadPackage(FileInfo packageFile, Palette pal)
```

Comment:
I'm not a fan of the palette being passed into all the package loaders when only the WAD provider needs it. I think it'd probably be better to cache the palette data inside the package by searching from the root like this:

```
if (sig == WAD2Signature) {
    var palette = packageFile.TraversePath("/gfx/palette.lmp");
    // Load palette byte data
}
```